packages/editable/src/use-editable.ts

```diff
@@ -159,7 +159,7 @@ export function useEditable(props: UseEditableProps = {}) {
   }, [value, onSubmitProp])
 
   const onChange = useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>) => {
+    (event: React.ChangeEvent<HTMLInputElement & HTMLTextAreaElement>) => {
```

Comment:
This could be misleading. The element type should somehow be inferred from the used `Element` and then either be `HTMLInputElement` or `HTMLTextAreaElement`

Reply:
First of all, Really thank you for review about subtle issue with nice manner.

Well, after a month of waiting other review,  I finally try to refactor this issue.
I wanna explain the reason I use union type as the solution on commit [f1bcf2c](https://github.com/chakra-ui/chakra-ui/pull/4443/commits/f1bcf2cd8d1765172665fda16af74a08c101bdc2)

1. On component `Editable`, there is no way to know which element will be used with the context. 
2. Because the props of editable were abstracted with just string input, so I think that it is reasonable not to reveal element type of change event.